### PR TITLE
Fix issues in scim filtering 

### PR DESCRIPTION
**Pull request by pamodaaw**
When the Login identifiers are defined, the SCIM filtering is not working for conditions that have more than one attribute. This PR fixes that issue.

Partially resolves, https://github.com/wso2/product-is/issues/10631

### CODE REVIEW

File: components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java [hunk 1]
```diff
@@ -1861,6 +1873,32 @@ private Condition getCondition(Node node, Map<String, String> attributes) throws
...
+                if (SCIMConstants.UserSchemaConstants.USER_NAME_URI.equals(attributeName) &&
+                        isLoginIdentifiersEnabled() && StringUtils.isNotBlank(getPrimaryLoginIdentifierClaim())) {
+                    attributeName = getScimUriForPrimaryLoginIdentifier(node);
+                }
+            } catch (org.wso2.carbon.user.core.UserStoreException e) {
+                throw new CharonException("Error in retrieving scim to local mappings.", e);
+            }
+
+
+            try {
```
Comment (Thumimku) [inline components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java:1889]: Is this logic redundant? the same logic is implemented inline 1876? WDYT?

Comment (pamodaaw) [reply components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java]: updated

**Review by Thumimku — APPROVED** [review]
LGTM

File: components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java [hunk 2]
```diff
@@ -1491,6 +1493,16 @@ private boolean isDomainNameEmbeddedInAttributeValue(String filterOperation, Str
             // Check for a user listing scenario. (For filtering this value will be set to NULL)
             if (conditionForListingUsers == null) {
 
+                if (isLoginIdentifiersEnabled()) {
+                    if (SCIMConstants.UserSchemaConstants.USER_NAME_URI
```
Comment (emswbandara) [inline components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java:1497]: shall we merge 2 if conditions?

File: components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java [hunk 3]
```diff
@@ -1861,6 +1873,18 @@ private Condition getCondition(Node node, Map<String, String> attributes) throws
             String attributeName = ((ExpressionNode) node).getAttributeValue();
             String attributeValue = ((ExpressionNode) node).getValue();
 
+            try {
+                /*If primary login identifier feature is enabled, the username uri should be replaced with
```
Comment (emswbandara) [inline components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java:1877]: formatting issue in comment

**Review by emswbandara — APPROVED** [review]

**Pull request merged**